curated_transformers/tokenization/hf_hub.py

```diff
@@ -73,78 +33,76 @@ def from_hf_hub(
-    The mixin will then provide the ``from_hf_tokenizer`` method to download a
-    tokenizer from the Hugging Face Hub.
+    Subclass of :class:`.FromHFHub` for legacy tokenizers. This subclass
+    implements the ``from_hf_hub`` method and provides through the abstract
+    ``_load_from_vocab_files`` method:
+
+    * The vocabulary files requested by a tokenizer through the
+      ``vocab_files`` member variable.
+    * The tokenizer configuration (when available).
     """
 
+    vocab_files: Dict[str, str] = {}
+
     @classmethod
     @abstractmethod
-    def _convert_hf_tokenizer(
-        cls: Type[SelfFromPretrainedTokenizer], tokenizer: Any
-    ) -> SelfFromPretrainedTokenizer:
+    def _load_from_vocab_files(
+        cls: Type[SelfLegacyFromHFHub],
+        *,
+        vocab_files: Dict[str, Path],
+        tokenizer_config: Optional[Dict[str, Any]]
+    ) -> SelfLegacyFromHFHub:
         """
-        Create an instance of the tokenizer from the Hugging Face
-        tokenizer instance.
+        Construct a tokenizer from its vocabulary files and optional
+        configuration.
 
-        :param tokenizer:
-            Hugging Face tokenizer.
+        :param vocab_files:
+            The resolved vocabulary files (in a local cache).
+        :param tokenizer_config:
+            The tokenizer configuration (when available).
         :returns:
-            New tokenizer.
+            The tokenizer.
         """
         raise NotImplementedError
 
     @classmethod
-    def from_hf_tokenizer(
-        cls: Type[SelfFromPretrainedTokenizer], *, name: str, revision: str = "main"
-    ) -> SelfFromPretrainedTokenizer:
-        """
-        Construct a tokenizer and load its parameters from Hugging Face Hub.
+    def from_hf_hub(
+        cls: Type[SelfLegacyFromHFHub], *, name: str, revision: str = "main"
+    ) -> SelfLegacyFromHFHub:
+        """Construct a tokenizer and load its parameters from Hugging Face Hub.
 
         :param name:
             Model name.
         :param revision:
             Model revision.
         :returns:
-            New tokenizer.
+            The tokenizer.
         """
-
-        # We cannot directly use the Hugging Face Hub downloader like we do for the model
-        # checkpoints as the tokenizer implementations do not use a consistent serialization
-        # interface with stable filenames for artifacts.
-        if not has_hf_transformers:
-            raise ValueError(
-                "`Loading tokenizers from Hugging Face Hub requires `transformers` package to be installed"
+        vocab_files = {}
+        for vocab_file, filename in cls.vocab_files.items():
+            vocab_files[vocab_file] = Path(
+                hf_hub_download(repo_id=name, filename=filename, revision=revision)
             )
 
-        tokenizer = transformers.AutoTokenizer.from_pretrained(name, revision=revision)
-        return cls._convert_hf_tokenizer(tokenizer)
+        # Try to get the tokenizer configuration.
```

Comment:
I wondering if it would be cleaner if we move the config downloading code to `util.hf` like we do with the model config and checkpoints.

Reply:
Fixed.